### PR TITLE
DM-194 "Hello World" CWU Page Structure

### PR DESCRIPTION
**Pull request by jjbandit**

### CODE REVIEW

**Comment by dhruvio** [comment]
The routes weren't working because they didn't start with a slash `/`

File: src/front-end/typescript/lib/app/router.ts [hunk 1]
```diff
@@ -31,7 +31,73 @@ export function back() {
...
+      path: '/code-with-us/opportunities/create',
+      makeRoute() {
+        return {
+          tag: 'cwuOpportunityCreate',
+          value: null
+        };
+      }
+    },
+    {
+      path: '/code-with-us/opportunities/edit',
```
Comment (dhruvio) [inline src/front-end/typescript/lib/app/router.ts]: Update to `/code-with-us/opportunities/:opportunityId/edit` and update the page to support an `opportunityId` route param

File: src/front-end/typescript/lib/app/router.ts [hunk 2]
```diff
@@ -31,7 +31,73 @@ export function back() {
...
+      makeRoute() {
+        return {
+          tag: 'cwuOpportunityEdit',
+          value: null
+        };
+      }
+    },
+    {
+      path: '/code-with-us/opportunities/view',
```
Comment (dhruvio) [inline src/front-end/typescript/lib/app/router.ts]: Update to `/code-with-us/opportunities/:opportunityId` (add ID & remove the `view` string) and update the page to support an `opportunityId` route param

File: src/front-end/typescript/lib/app/router.ts [hunk 3]
```diff
@@ -31,7 +31,73 @@ export function back() {
...
+      path: '/code-with-us/proposals/create',
+      makeRoute() {
+        return {
+          tag: 'cwuProposalCreate',
+          value: null
+        };
+      }
+    },
+    {
+      path: '/code-with-us/proposals/edit',
```
Comment (dhruvio) [inline src/front-end/typescript/lib/app/router.ts]: Update to `/code-with-us/proposals/:proposalId` and update the page to support a `proposalId` route param

File: src/front-end/typescript/lib/app/router.ts [hunk 4]
```diff
@@ -31,7 +31,73 @@ export function back() {
...
+      makeRoute() {
+        return {
+          tag: 'cwuProposalEdit',
+          value: null
+        };
+      }
+    },
+    {
+      path: '/code-with-us/proposals/view',
```
Comment (dhruvio) [inline src/front-end/typescript/lib/app/router.ts]: Update to `/code-with-us/proposals/:proposalId` (add ID & remove the `view` string) and update the page to support a `proposalId` route param

File: src/front-end/typescript/lib/app/router.ts [hunk 5]
```diff
@@ -31,7 +31,73 @@ export function back() {
...
+      makeRoute() {
+        return {
+          tag: 'cwuProposalView',
+          value: null
+        };
+      }
+    },
+    {
+      path: '/code-with-us/proposals/list',
```
Comment (dhruvio) [inline src/front-end/typescript/lib/app/router.ts]: Remove the `list` suffix

Comment (dhruvio) [reply src/front-end/typescript/lib/app/router.ts]: Also, per the ticket, this should just  be `/proposals`

File: src/front-end/typescript/lib/app/router.ts [hunk 6]
```diff
@@ -31,7 +31,73 @@ export function back() {
...
+      makeRoute() {
+        return {
+          tag: 'cwuProposalList',
```
Comment (dhruvio) [inline src/front-end/typescript/lib/app/router.ts]: Rename to `proposalList`

File: src/front-end/typescript/lib/app/router.ts [hunk 7]
```diff
@@ -261,19 +261,19 @@ const router: Router<Route> = {
...
+        return '/proposals/code-with-us/create';
       case 'cwuProposalEdit':
-        return 'code-with-us/proposals/edit';
+        return '/proposals/code-with-us/edit';
       case 'cwuProposalView':
-        return 'code-with-us/proposals/view';
+        return '/proposals/code-with-us/view';
       case 'cwuProposalList':
-        return 'code-with-us/proposals/list';
+        return '/proposals/code-with-us';
```
Comment (dhruvio) [inline src/front-end/typescript/lib/app/router.ts]: This should be `/proposals`

File: src/front-end/typescript/lib/app/router.ts (hunk 6 above)
Comment (dhruvio) [reply src/front-end/typescript/lib/app/router.ts]: See the above comment

File: src/front-end/typescript/lib/app/router.ts [hunk 8]
```diff
@@ -31,7 +31,73 @@ export function back() {
...
+        };
+      }
+    },
+    {
+      path: '/opportunities/code-with-us/:id/edit',
+      makeRoute() {
+        return {
+          tag: 'cwuOpportunityEdit',
+          value: null
+        };
```
Comment (dhruvio) [inline src/front-end/typescript/lib/app/router.ts]: Pass ID to route params

File: src/front-end/typescript/lib/app/router.ts [hunk 9]
```diff
@@ -31,7 +31,73 @@ export function back() {
...
+      }
+    },
+    {
+      path: '/opportunities/code-with-us/:id',
+      makeRoute() {
+        return {
+          tag: 'cwuOpportunityView',
+          value: null
+        };
```
Comment (dhruvio) [inline src/front-end/typescript/lib/app/router.ts]: Pass ID to route params

File: src/front-end/typescript/lib/app/router.ts [hunk 10]
```diff
@@ -31,7 +31,73 @@ export function back() {
...
+      }
+    },
+    {
+      path: '/proposals/code-with-us/:id/edit',
+      makeRoute() {
+        return {
+          tag: 'cwuProposalEdit',
+          value: null
+        };
```
Comment (dhruvio) [inline src/front-end/typescript/lib/app/router.ts]: Pass ID to route params

File: src/front-end/typescript/lib/app/router.ts [hunk 11]
```diff
@@ -31,7 +31,73 @@ export function back() {
...
+      }
+    },
+    {
+      path: '/proposals/code-with-us/:id',
+      makeRoute() {
+        return {
+          tag: 'cwuProposalView',
+          value: null
+        };
```
Comment (dhruvio) [inline src/front-end/typescript/lib/app/router.ts]: Pass ID to route params

File: src/front-end/typescript/lib/app/router.ts [hunk 12]
```diff
@@ -192,6 +260,20 @@ const router: Router<Route> = {
         return `/organizations/${route.value.orgId}/edit`;
       case 'orgCreate':
         return '/organizations/create';
+      case 'cwuOpportunityCreate':
+        return '/code-with-us/opportunities/create';
+      case 'cwuOpportunityEdit':
+        return '/code-with-us/opportunities/edit';
```
Comment (dhruvio) [inline src/front-end/typescript/lib/app/router.ts]: Encode ID in URL

File: src/front-end/typescript/lib/app/router.ts [hunk 13]
```diff
@@ -192,6 +260,20 @@ const router: Router<Route> = {
...
+      case 'cwuOpportunityView':
+        return '/code-with-us/opportunities/view';
```
Comment (dhruvio) [inline src/front-end/typescript/lib/app/router.ts]: Encode ID in URL and remove `view`

File: src/front-end/typescript/lib/app/router.ts [hunk 14]
```diff
@@ -192,6 +260,20 @@ const router: Router<Route> = {
...
+      case 'cwuProposalCreate':
+        return '/proposals/code-with-us/create';
+      case 'cwuProposalEdit':
+        return '/proposals/code-with-us/edit';
```
Comment (dhruvio) [inline src/front-end/typescript/lib/app/router.ts]: Encode ID in URL

File: src/front-end/typescript/lib/app/router.ts [hunk 15]
```diff
@@ -192,6 +260,20 @@ const router: Router<Route> = {
...
+      case 'cwuProposalView':
+        return '/proposals/code-with-us/view';
```
Comment (dhruvio) [inline src/front-end/typescript/lib/app/router.ts]: Encode ID in URL and remove `view`

File: src/front-end/typescript/lib/app/types.ts [hunk 16]
```diff
@@ -71,25 +88,40 @@ export interface State {
     orgCreate?: Immutable<PageOrgCreate.State>;
     orgList?: Immutable<PageOrgList.State>;
     orgEdit?: Immutable<PageOrgEdit.State>;
+    cwuOpportunityCreate?: Immutable<CwuOpportunityCreate.State>;
+    cwuOpportunityEdit?: Immutable<CwuOpportunityEdit.State>;
+    cwuOpportunityView?: Immutable<CwuOpportunityView.State>;
+    cwuProposalCreate?: Immutable<CwuProposalCreate.State>;
+    cwuProposalEdit?: Immutable<CwuProposalEdit.State>;
+    cwuProposalView?: Immutable<CwuProposalView.State>;
+    cwuProposalList?: Immutable<CwuProposalList.State>;
```
Comment (dhruvio) [inline src/front-end/typescript/lib/app/types.ts]: Change to `proposalList`

File: src/front-end/typescript/lib/app/types.ts [hunk 17]
```diff
@@ -17,20 +25,28 @@ import { includes } from 'lodash';
...
+  | ADT<'orgCreate',             PageOrgCreate.RouteParams>
+  | ADT<'orgList',               PageOrgList.RouteParams>
+  | ADT<'orgEdit',               PageOrgEdit.RouteParams>
+  | ADT<'cwuOpportunityCreate',  CwuOpportunityCreate.RouteParams>
+  | ADT<'cwuOpportunityEdit',    CwuOpportunityEdit.RouteParams>
+  | ADT<'cwuOpportunityView',    CwuOpportunityView.RouteParams>
+  | ADT<'cwuProposalCreate',     CwuProposalCreate.RouteParams>
+  | ADT<'cwuProposalEdit',       CwuProposalEdit.RouteParams>
+  | ADT<'cwuProposalView',       CwuProposalView.RouteParams>
+  | ADT<'cwuProposalList',       CwuProposalList.RouteParams>;
```
Comment (dhruvio) [inline src/front-end/typescript/lib/app/types.ts]: Change to `proposalList`

File: src/front-end/typescript/lib/app/types.ts [hunk 18]
```diff
@@ -1,6 +1,14 @@
 import * as Nav from 'front-end/lib/app/view/nav';
 import { AppMsg, Immutable, PageModal } from 'front-end/lib/framework';
 import * as PageContent from 'front-end/lib/pages/content';
+// Note(Jesse): @add_new_page_location
+import * as CwuOpportunityCreate from 'front-end/lib/pages/cwu/opportunities/create';
+import * as CwuOpportunityEdit from 'front-end/lib/pages/cwu/opportunities/edit';
+import * as CwuOpportunityView from 'front-end/lib/pages/cwu/opportunities/view';
+import * as CwuProposalCreate from 'front-end/lib/pages/cwu/proposals/create';
+import * as CwuProposalEdit from 'front-end/lib/pages/cwu/proposals/edit';
+import * as CwuProposalList from 'front-end/lib/pages/cwu/proposals/list';
```
Comment (dhruvio) [inline src/front-end/typescript/lib/app/types.ts]: Change to `proposalList` and move module accordingly

File: src/front-end/typescript/lib/app/types.ts [hunk 19]
```diff
@@ -71,25 +88,40 @@ export interface State {
...
+  | ADT<'pageOrgCreate',         PageOrgCreate.Msg>
+  | ADT<'pageOrgList',           PageOrgList.Msg>
+  | ADT<'pageOrgEdit',           PageOrgEdit.Msg>
+  | ADT<'cwuOpportunityCreate',  CwuOpportunityCreate.Msg>
+  | ADT<'cwuOpportunityEdit',    CwuOpportunityEdit.Msg>
+  | ADT<'cwuOpportunityView',    CwuOpportunityView.Msg>
+  | ADT<'cwuProposalCreate',     CwuProposalCreate.Msg>
+  | ADT<'cwuProposalEdit',       CwuProposalEdit.Msg>
+  | ADT<'cwuProposalView',       CwuProposalView.Msg>
+  | ADT<'cwuProposalList',       CwuProposalList.Msg>;
```
Comment (dhruvio) [inline src/front-end/typescript/lib/app/types.ts]: All page messages should be prefixed with `page` to match the other page messages

Comment (dhruvio) [reply src/front-end/typescript/lib/app/types.ts]: Change to `cwuProposalList` to `pageProposalList`

**Comment by jjbandit** [comment]
@dhruvio  All changes addressed I believe

**Pull request merged**